react/features/recording/googleApi/ElectronGoogleClient.js

```diff
@@ -0,0 +1,269 @@
+     * user associated with the {@link accessToken}.
+     *
+     * @returns {Promise}
+     */
+    requestAvailableYouTubeBroadcasts() {
+        const requestUrl = this._getURLForLiveBroadcasts({
+            // eslint-disable-next-line camelcase
+            access_token: this._getAccessToken()
+        });
+
+        return fetch(requestUrl)
+            .then(res => this._mimicGoogleWebClientResponse(res));
+    }
+
+    /**
+     * Executes a request to get all live streams associated with a broadcast
+     * in YouTube.
+     *
+     * @param {string} boundStreamID - The bound stream ID associated with a
+     * broadcast in YouTube.
+     * @returns {Promise}
+     */
+    requestLiveStreamsForYouTubeBroadcast(boundStreamID: string) {
+        const requestUrl = this._getURLForLiveStreams({
+            // eslint-disable-next-line camelcase
+            access_token: this._getAccessToken(),
+            id: boundStreamID
+        });
+
+        return fetch(requestUrl)
+            .then(res => this._mimicGoogleWebClientResponse(res));
+    }
+
+    /**
+     * Prompts the participant to sign in Google, even if a valid access token
+     * is known.
+     *
+     * @returns {Promise}
+     */
+    showAccountSelection() {
+        const redirectUri = this._getRedirectUri();
+        const authUrl
+            = `https://accounts.google.com/o/oauth2/v2/auth?client_id=${
+                clientId}&redirect_uri=${redirectUri}&scope=${
+                this._getApiScope()}&response_type=token`;
+
+        return this._getGoogleApiClient()
+            .requestToken({
+                authUrl,
+                redirectUri
+            })
+            .then(url => this._parseAccessToken(url))
+            .then(token =>
+                this._validateAccessToken(token)
+                    .then(isValid => {
+                        if (isValid) {
+                            this._setAccessToken(token);
+
+                            return;
+                        }
+
+                        return Promise.reject();
+                    }));
+    }
+
+    /**
+     * Helper method to access the last known valid access token.
+     *
+     * @private
+     * @returns {string}
+     */
+    _getAccessToken() {
+        return window.sessionStorage.getItem(sessionStorageKey);
+    }
+
+    _getApiScope: () => string;
+
+    /**
+     * Returns the global object from the Electron environment which allows
+     * communication with Google oauth.
+     *
+     * @private
+     * @returns {Object}
+     */
+    _getGoogleApiClient(): Object {
+        return JitsiMeetElectron.googleApi;
+    }
+
+    /**
+     * Helper method to generate the redirect URI Google should visit after
+     * successful authentication.
+     *
+     * @private
+     * @returns {string}
+     */
+    _getRedirectUri() {
+        return `${window.location.origin}/static/googleAuthRedirect.html`;
+    }
+
+    _getURLForLiveBroadcasts: (Object) => string;
```

Comment:
double check

Reply:
Looks like I added these to flow. The subclass is calling them but they're declared on the base class and flow yells if I don't declare the signature in the subclass.